Assets/MRTK/Core/Services/MixedRealityToolkit.cs

```diff
@@ -88,14 +93,38 @@ public MixedRealityToolkitConfigurationProfile ActiveProfile
             }
             set
             {
-                ResetConfiguration(value);
+                if (Application.isPlaying && activeProfile != null && value != null && activeProfile != value)
+                {
+                    newProfile = value;
+                }
+                else
+                {
+                    ResetConfiguration(value);
+                }
+                
             }
         }
 
         /// <summary>
         /// When a configuration Profile is replaced with a new configuration, force all services to reset and read the new values
         /// </summary>
         public void ResetConfiguration(MixedRealityToolkitConfigurationProfile profile)
+        {
+            RemoveCurrentProfile(profile);
+            InitializeNewProfile(profile);
+        }
+
+        private void InitializeNewProfile(MixedRealityToolkitConfigurationProfile profile)
+        {
+            InitializeServiceLocator();
+
+            if (profile != null && Application.IsPlaying(profile))
+            {
+                EnableAllServices();
+            }
+        }
+
+        private void RemoveCurrentProfile(MixedRealityToolkitConfigurationProfile profile)
```

Comment:
It's also a bit weird here that this function sets activeProfile = profile (i.e. the act of setting it calls that again) - is that call still necessary here?

Reply:
Nevermind, I thought I saw a capital A there.